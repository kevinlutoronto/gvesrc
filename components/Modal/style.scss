@import '../../styles/base';

@keyframes slidein {
  from {
    opacity: .5;
    top: 20px;
  }

  to {
    opacity: 1;
    top: 0;
  }
}

@keyframes fadeOverlay {
  from {
    opacity: .5;
  }

  to {
    opacity: 1;
  }
}

.wrapper {
  position: fixed;
  left: 0;
  top: 0;
  right: 0;
  bottom: 0;
  z-index: 100;
}

.innerWrapper {
  position: absolute;
  left: 0;
  right: 0;
  top: 0;
  bottom: 0;
  overflow-y: scroll;
  display: flex;
  justify-content: center;
  align-items: flex-start;
}

.overlay {
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  background: rgba($cPurple100, 0.94);
  z-index: 101;
  animation-duration: 0.15s;
  animation-name: fadeOverlay;
}

.card {
  position: relative;
  background: $cWhite;
  box-shadow: rgba(0,0,0,0.3) 0 5px 10px;
  z-index: 102;
  border-radius: 5px;
  animation-duration: 0.15s;
  animation-name: slidein;
  animation-timing-function: ease-out;
  margin: 40px 0;
}

.content {
  @extend %layout--bounds;
}

.pageActions {
  display: flex;

  > * + * {
    margin-left: 20px;
  }
}
