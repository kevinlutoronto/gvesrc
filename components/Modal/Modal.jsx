// @flow
import React, { type Node } from 'react';
import cn from 'classnames';
import style from './style.scss';
import Header from '../Header/Header';
import { getDataAttrs } from '../../dataUtils';
import type { Data, StyleAttributes } from '../../types';

type Props = {|
  title?: string,
  actions?: Array<Node>,
  children: React$Node,
  visible?: boolean,
  onDismiss?: (evt: MouseEvent) => any,
  innerCardStyle?: StyleAttributes,
  minWidth?: string | number,
  maxWidth?: string | number,
  data?: Data,
  contentClassName?: string,
|};

class Modal extends React.Component<Props> {
  render() {
    const {
      title,
      actions,
      data,
      children,
      onDismiss,
      minWidth = 0,
      maxWidth,
      visible,
      innerCardStyle,
      contentClassName,
    } = this.props;

    if (!visible) {
      return null;
    }

    return (
      <div {...getDataAttrs(data)} className={style.wrapper}>
        <div className={style.innerWrapper}>
          <div style={{ ...innerCardStyle, minWidth, maxWidth }} className={style.card}>
            <div className={cn(style.content, contentClassName)}>
              {title && (
                <div className={style.top}>
                  <Header h3>{title}</Header>
                </div>
              )}
              {children}
              {actions && <div className={style.pageActions}>{actions}</div>}
            </div>
          </div>
        </div>
        <div className={style.overlay} onClick={onDismiss} />
      </div>
    );
  }
}

Modal.displayName = 'Plasma@Modal';

export default Modal;
