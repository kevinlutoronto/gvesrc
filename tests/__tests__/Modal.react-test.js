/* eslint-disable */

import React from 'react';
import Modal from '../../components/Modal/Modal';
import renderer from 'react-test-renderer';
import { shallow, mount } from 'enzyme';

describe('Modal Component', () => {
  test('render', () => {
    const component = renderer.create(<Modal>YOLO</Modal>);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
