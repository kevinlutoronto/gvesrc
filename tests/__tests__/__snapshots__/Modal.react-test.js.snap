// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Component render 1`] = `null`;
